### PR TITLE
AP_Baro: fix instance shadowing

### PR DESCRIPTION
**Pull request by khancyr**
and minor style fix : space, f qualifier, const, use is_positive for float

### CODE REVIEW

**Comment by magicrub** [comment]
this looks fine, but unnecessary..
- the instance rename: is there a problem with the current name?
- float being set by an integer is fine, casting it to a float first does nothing useful

**Comment by khancyr** [comment]
@magicrub Yep, if you look at current build you should see 
`In file included from ../../libraries/AP_Baro/AP_Baro.cpp:31:0:
../../libraries/AP_Baro/AP_Baro_SITL.h: In member function ‘virtual void AP_Baro_SITL::update_healthy_flag(uint8_t)’:
../../libraries/AP_Baro/AP_Baro_SITL.h:17:57: warning: declaration of ‘instance’ shadows a member of ‘AP_Baro_SITL’ [-Wshadow]
     void update_healthy_flag(uint8_t instance) override { _frontend.sensors[instance].healthy = true; }
                                                         ^
../../libraries/AP_Baro/AP_Baro_SITL.h:20:13: note: shadowed declaration is here
     uint8_t instance;
`

Where did I cast to float ? If it is about the f qualifier, it is mostly just to be rigorous, but sometime it allow to see directly that we are doing floaf calculus. So it cost nothing to add it everywhere it should be !

File: libraries/AP_Baro/AP_Baro_SITL.cpp [hunk 1]
```diff
@@ -55,29 +55,29 @@ void AP_Baro_SITL::_timer()
...
-    uint8_t best_index = 0; // initialise number representing the index of the entry in buffer closest to delay.
+    uint32_t best_time_delta = 200;  // initialise large time representing buffer entry closest to current time - delay.
+    uint8_t best_index = 0;  // initialise number representing the index of the entry in buffer closest to delay.
 
     // storing data from sensor to buffer
-    if (now - last_store_time >= 10) { // store data every 10 ms.
+    if (now - last_store_time >= 10) {  // store data every 10 ms.
         last_store_time = now;
-        if (store_index > buffer_length-1) { // reset buffer index if index greater than size of buffer
+        if (store_index > buffer_length-1) {  // reset buffer index if index greater than size of buffer
```
Comment (muramura) [inline libraries/AP_Baro/AP_Baro_SITL.cpp]: I do not know why why white space is not included in this process.

File: libraries/AP_Baro/AP_Baro_SITL.cpp [hunk 2]
```diff
@@ -55,29 +55,29 @@ void AP_Baro_SITL::_timer()
...
+        buffer[store_index].time = last_store_time;  // add time_stamp to current index
+        store_index = store_index + 1;  // increment index
     }
 
     // return delayed measurement
-    uint32_t delayed_time = now - sitl->baro_delay; // get time corresponding to delay
+    const uint32_t delayed_time = now - sitl->baro_delay;  // get time corresponding to delay
 
     // find data corresponding to delayed time in buffer
     for (uint8_t i=0; i<=buffer_length-1; i++) {
```
Comment (muramura) [inline libraries/AP_Baro/AP_Baro_SITL.cpp]: I do not know why why white space is not included in this process.

File: libraries/AP_Baro/AP_Baro_SITL.cpp [hunk 3]
```diff
@@ -55,29 +55,29 @@ void AP_Baro_SITL::_timer()
...
-        if (store_index > buffer_length-1) { // reset buffer index if index greater than size of buffer
+        if (store_index > buffer_length-1) {  // reset buffer index if index greater than size of buffer
             store_index = 0;
         }
-        buffer[store_index].data = sim_alt; // add data to current index
-        buffer[store_index].time = last_store_time; // add time_stamp to current index
-        store_index = store_index + 1; // increment index
+        buffer[store_index].data = sim_alt;  // add data to current index
+        buffer[store_index].time = last_store_time;  // add time_stamp to current index
+        store_index = store_index + 1;  // increment index
```
Comment (muramura) [inline libraries/AP_Baro/AP_Baro_SITL.cpp]: I write it as increment in comments. I think that "++" is good.

File: libraries/AP_Baro/AP_Baro_SITL.h [hunk 4]
```diff
@@ -28,7 +28,7 @@ class AP_Baro_SITL : public AP_Baro_Backend {
     uint8_t store_index;
```
Comment (muramura) [inline libraries/AP_Baro/AP_Baro_SITL.h]: I can not understand why "_" is not attached to this private variable.

**Comment by lucasdemarchi** [comment]
Thank you, pull request merged!

**Pull request closed without merging**